test/-ext-/thread/test_instrumentation_api.rb

```diff
@@ -4,17 +4,23 @@ def setup
     pend("TODO: No windows support yet") if /mswin|mingw|bccwin/ =~ RUBY_PLATFORM
   end
 
+  THREADS_COUNT = 3
```

Comment:
I think it may be because CI runners don't have many cores, so 5 threads might have caused some to be waiting to be natively scheduled to execute the exit hook.

3 should be more than enough for our assertions anyway.